model/implementation/DeliveryExecutionStateService.php

```diff
@@ -216,6 +213,12 @@ public function terminateExecution(DeliveryExecution $deliveryExecution, $reason
                 $this->getTestSessionService()->persist($session);
                 $this->getServiceLocator()->get(ExtendedStateService::SERVICE_ID)->persist($session->getSessionId());
             }
+            
+            // Delivery execution state changes after test session ends, in the same way as it happens
+            // when a human test taker takes the test.
+            $eventManager->trigger(new DeliveryExecutionTerminated($deliveryExecution, $proctor, $reason));
```

Comment:
@jsconan  **Be very careful here ! I changed the order of the events being triggered.** @antoinerobin and @bugalot do not think that's a problem, but it's worth to triple check it does not break anything elsewhere.

Reply:
Should not be a problem. I checked the use of this event and the order does not matter.